python/examples/advanced_agents/slackbot_code_expert/chat/main.py

```diff
@@ -0,0 +1,52 @@
+from dotenv import load_dotenv
+load_dotenv()
+import os
+import sys
+from openai import OpenAI
+from db import ChatDB
+
+from .codebase_agent import find_code_snippet
+from .tools import code_search_tool
+
+# Add the parent directory to sys.path
+current_dir = os.path.dirname(os.path.abspath(__file__))
+parent_dir = os.path.dirname(current_dir)
+sys.path.append(parent_dir)
+
+client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))
+
+chat_db = ChatDB(db_path=os.path.join(parent_dir, 'db', 'db.json'))
+
+def chatbot(messages: list):
+    # Users query is passed in the last message, if it requires more context to answer the question it uses code_search_tool to find the relevant code snippet
+    completion = client.chat.completions.create(
+        model="gpt-4o-mini",
+        messages=messages,
+        tools=code_search_tool,
+    )
+    # Checking if response from above completion contains tool calls, if so it uses the tool to find the code snippet
+    if completion.choices[0].message.tool_calls:
+        tool_call = completion.choices[0].message.tool_calls[0]
+        function_name = tool_call.function.name
+        arguments = eval(tool_call.function.arguments)
```

Comment:
The `eval` function is used here to parse arguments (line 31). Using `eval` can introduce security risks, especially with untrusted input. Consider using a safer alternative for parsing the arguments, such as `ast.literal_eval` or a custom parsing function.

Reply:
Got it! Will use the alternative